Source/WebCore/platform/graphics/AV1Utilities.cpp

```diff
@@ -596,16 +614,758 @@ std::optional<AV1CodecConfigurationRecord> parseAV1DecoderConfigurationRecord(co
     else
         record.bitDepth = 8;
 
-    record.monochrome = tierBitdepthAndColorFlags & 0b00010000;
-    uint8_t chromaSubsamplingValue = 0;
-    if (tierBitdepthAndColorFlags & 0b00001000)
-        chromaSubsamplingValue += 100;
-    if (tierBitdepthAndColorFlags & 0b00000100)
-        chromaSubsamplingValue += 10;
-    chromaSubsamplingValue += tierBitdepthAndColorFlags & 0b00000010;
+    // monochrome f(1)
+    value = bitReader.read(1);
+    if (!value)
+        return std::nullopt;
+    record.monochrome = *value;
+
+    // chroma_subsampling_x f(1)
+    value = bitReader.read(1);
+    if (!value)
+        return std::nullopt;
+    uint8_t chromaSubsamplingX = *value;
+
+    // chroma_subsampling_y f(1)
+    value = bitReader.read(1);
+    if (!value)
+        return std::nullopt;
+    uint8_t chromaSubsamplingY = *value;
+
+    // chroma_sample_position f(2)
+    value = bitReader.read(2);
+    if (!value)
+        return std::nullopt;
+    uint8_t chromaSamplePosition = static_cast<uint8_t>(*value);
+
+    // Compute chromaSubsampling value: first digit = subsampling_x, second = subsampling_y, third = chroma_sample_position (if 4:2:0) or 0
+    record.chromaSubsampling = chromaSubsamplingX * 100 + chromaSubsamplingY * 10 + ((chromaSubsamplingX && chromaSubsamplingY) ? chromaSamplePosition : 0);
+
+    // Initialize dimension fields to default values (decoder config doesn't contain dimensions)
+    record.width = AV1CodecConfigurationRecord::defaultWidth;
+    record.height = AV1CodecConfigurationRecord::defaultHeight;
+    record.codecName = "av01"_s;
+
+    return record;
+}
+
+std::optional<AV1CodecConfigurationRecord> parseSequenceHeaderOBU(std::span<const uint8_t> data)
+{
+    AV1CodecConfigurationRecord record;
+    record.codecName = "av01"_s;
+
+    BitReader bitReader(data);
+    std::optional<size_t> value;
+
+    // seq_profile f(3)
```

Comment:
What's the strategy for testing and/or fuzzing these binary parsers?

Reply:
> What's the strategy for testing and/or fuzzing these binary parsers?

all media parsers have been moved to the web content process now.